rust/worker/src/config.rs

```diff
@@ -0,0 +1,179 @@
+use figment::providers::{Env, Format, Serialized, Yaml};
+use serde::Deserialize;
+
+const DEFAULT_CONFIG_PATH: &str = "chroma_config.yaml";
+const ENV_PREFIX: &str = "CHROMA_";
+
+#[derive(Deserialize)]
+/// # Description
+/// The RootConfig for all chroma services this is a YAML file that
+/// is shared between all services, and secondarily, fields can be
+/// populated from environment variables. The environment variables
+/// are prefixed with CHROMA_ and are uppercase. Values in the envionment
+/// variables take precedence over values in the YAML file.
+/// By default, it is read from the current working directory,
+/// with the filename chroma_config.yaml.
+struct RootConfig {
+    // The root config object wraps the worker config object so that
+    // we can share the same config file between multiple services.
+    worker: WorkerConfig,
+}
+
+impl RootConfig {
+    /// # Description
+    /// Load the config from the default location.
+    /// # Returns
+    /// The config object.
+    /// # Panics
+    /// - If the config file cannot be read.
+    /// - If the config file is not valid YAML.
+    /// - If the config file does not contain the required fields.
+    /// - If the config file contains invalid values.
+    /// - If the environment variables contain invalid values.
+    /// # Notes
+    /// The default location is the current working directory, with the filename chroma_config.yaml.
+    /// The environment variables are prefixed with CHROMA_ and are uppercase.
+    /// Values in the envionment variables take precedence over values in the YAML file.
+    pub fn load() -> Self {
+        return Self::load_from_path(DEFAULT_CONFIG_PATH);
+    }
+
+    /// # Description
+    /// Load the config from a specific location.
+    /// # Arguments
+    /// - path: The path to the config file.
+    /// # Returns
+    /// The config object.
+    /// # Panics
+    /// - If the config file cannot be read.
+    /// - If the config file is not valid YAML.
+    /// - If the config file does not contain the required fields.
+    /// - If the config file contains invalid values.
+    /// - If the environment variables contain invalid values.
+    /// # Notes
+    /// The environment variables are prefixed with CHROMA_ and are uppercase.
+    /// Values in the envionment variables take precedence over values in the YAML file.
+    pub fn load_from_path(path: &str) -> Self {
+        // Unfortunately, figment doesn't support environment variables with underscores. So we have to map and replace them.
```

Comment:
This is brittle and a little clunky, but I think its our only option for now and since this is all self contained in one file I am ok with it.